File: KLASA_3/ROZDZIAL_1/LEKCJA_1/C++/ZAD3.cpp
#include <iostream>
#include <stack>
#include <math.h>

using namespace std;

int main()
{
    stack<int> stos;

    string onp;
    cout << "Podaj wyrazenie zapisane w ONP (np. 23+5* [to (2+3)*5]): ";
    cin >> onp;

    for(int i=0;i<onp.size();i++){
        if (onp[i] >= '1' && onp[i] <= '9'){
            stos.push(int(onp[i])-'0');
        }else{
            int arg2 = stos.top();
            stos.pop();
            int arg1 = stos.top();
            stos.pop();

            char znak = onp[i];
            if(znak == '+'){
                stos.push(arg1+arg2);
            }else if(znak == '-'){
                stos.push(arg1-arg2);
            }else if(znak == '*'){
                stos.push(arg1*arg2);
            }else if(znak == '/'){
                stos.push(arg1/arg2);
            }else if(znak == '%'){
                stos.push(arg1%arg2);
            }else if(znak == '^'){
                stos.push(pow(arg1,arg2));
            }
        }
    }

    cout << stos.top();

    return 0;
}

File: KLASA_3/ROZDZIAL_1/LEKCJA_1/PYTHON/ZAD3.py
from stos import Stos


def wartosc_onp(postfix):
    stos = Stos()
    for i in range(len(postfix)):
        if postfix[i] >= '1' and postfix[i] <= '9':
            stos.push(int(postfix[i]))
        else:
            arg2 = stos.peek()
            stos.pop()
            arg1 = stos.peek()
            stos.pop()

            znak = postfix[i]
            if znak == '+':
                stos.push(arg1 + arg2)
            elif znak == '-':
                stos.push(arg1 - arg2)
            elif znak == '*':
                stos.push(arg1 * arg2)
            elif znak == '/':
                stos.push(arg1 / arg2)
            elif znak == '%':
                stos.push(arg1 % arg2)
            elif znak == '^':
                stos.push(arg1 ** arg2)
    wynik = stos.peek()
    return wynik


print(wartosc_onp(input('Podaj wyrażenie zapisane w ONP (np. 23+5* [to (2+3)*5]): ')))